internal/server/iigointernal/judiciary.go

```diff
@@ -132,3 +151,140 @@ func softMergeSanctionThresholds(clientSanctionMap map[roles.IIGOSanctionTier]ro
 	}
 	return defaultMap
 }
+
+// getIslandSanctionTier if statement based evaluator for which sanction tier a particular score is in
+func getIslandSanctionTier(islandScore roles.IIGOSanctionScore, scoreMap map[roles.IIGOSanctionTier]roles.IIGOSanctionScore) roles.IIGOSanctionTier {
+	if islandScore <= scoreMap[roles.SanctionTier1] {
+		return roles.NoSanction
+	} else if islandScore <= scoreMap[roles.SanctionTier2] {
+		return roles.SanctionTier1
+	} else if islandScore <= scoreMap[roles.SanctionTier3] {
+		return roles.SanctionTier2
+	} else if islandScore <= scoreMap[roles.SanctionTier4] {
+		return roles.SanctionTier3
+	} else if islandScore <= scoreMap[roles.SanctionTier5] {
+		return roles.SanctionTier4
+	} else {
+		return roles.SanctionTier5
+	}
+}
+
+// getTierSanctionMap basic mapping between snaciotn tier and rule that governs it
+func getTierSanctionMap() map[roles.IIGOSanctionTier]string {
+	return map[roles.IIGOSanctionTier]string{
+		roles.SanctionTier1: "iigo_economic_sanction_1",
+		roles.SanctionTier2: "iigo_economic_sanction_2",
+		roles.SanctionTier3: "iigo_economic_sanction_3",
+		roles.SanctionTier4: "iigo_economic_sanction_4",
+		roles.SanctionTier5: "iigo_economic_sanction_5",
+	}
+}
+
+// defaultInitLocalSanctionCache generates a blank sanction cache
+func defaultInitLocalSanctionCache(depth int) map[int][]roles.Sanction {
+	returnMap := map[int][]roles.Sanction{}
+	for i := 0; i < depth; i++ {
+		returnMap[i] = []roles.Sanction{}
+	}
+	return returnMap
+}
+
+func implementPardons(sanctionCache map[int][]roles.Sanction, pardons map[int][]bool, allTeamIds []shared.ClientID) (bool, map[int][]roles.Sanction, map[shared.ClientID][]map[shared.CommunicationFieldName]shared.CommunicationContent) {
+	if validatePardons(sanctionCache, pardons) {
+		finalSanctionCache := sanctionCache
```

Comment:
once again since this is simply a reference meaning  we're both iterating over and changing sanctionCache in the loop below there may be some unintended behaviour

Also would it not be easier for Sanctions to be a map of clientIDs to sanctions tiers and turns left. Unless of course a client can get multiple sanctions.

Reply:
In this case I've rewritten this function with that in mind, so everything is an assignment now hopefully. As before this map is indexed by timestep, not clientID.

Reply:
Ah the second part of my comment was more to do with the type of sanction. Its a struct of ClientID and then some sanction data. Would it not be easier to type it as a map and then not need a list i.e []roles.Sanctions -> roles.Sanctions(which is now a map)

Reply:
Hmmmm that's a good thought. Could we do that in another PR, this rebasing thing hasn't gone well.

Reply:
Sure, as long we have a list of all changes that need to be made. This one is also a lower priority.